packages/atomic/src/components/search/search-box-suggestions/suggestions-common.ts

```diff
@@ -3,41 +3,136 @@ import {VNode} from '@stencil/core';
+ */
 export interface SearchBoxSuggestionElement {
+  /**
+   * Stable identity which enables Stencil to reuse DOM elements for better performance.
+   * The best way to pick a key is to use a string that uniquely identifies that list item among its siblings (often your data will already have IDs).
+   */
   key: string;
+  /**
+   * Rendered content of the element.
+   */
   content: Element | VNode;
+  /**
+   * Hook called when the selection is selected.
+   * @param e DOM event.
+   */
   onSelect?(e: Event): void;
-
+  /**
+   * The query associated with the suggestion, which will become the suggested query that would be sent if the search is executed.
+   */
   query?: string;
+  /**
+   * For improved accessibility, provide this property with additional information.
+   * https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
+   */
   ariaLabel?: string;
+  /**
+   * Adds a specific shadow part attribute that can be selected with the CSS ::part pseudo-element.
+   * https://developer.mozilla.org/en-US/docs/Web/CSS/::part
+   */
   part?: string;
+  /**
+   * Hide the suggestion if it's the last in the list.
+   */
   hideIfLast?: boolean;
 }
+
+/**
+ * List of suggestions that will be displayed along other lists (e.g recent queries) when the search box's input is selected.
+ */
 export interface SearchBoxSuggestions {
+  /**
+   * The position of the list of the suggestions relative to the others.
+   */
   position: number;
+  /**
+   * Whether the suggestions should be listed in the right or left panel.
+   */
   panel?: 'left' | 'right';
+  /**
+   * Method that returns the list of elements which will be rendered in the list of suggestions.
+   */
   renderItems(): SearchBoxSuggestionElement[];
+  /**
+   * Hook called when the user changes the search box's input value.
+   */
   onInput?(): Promise<unknown> | void;
+  /**
+   * Hook called when the suggested query changes.
+   * @param q The new suggested query.
+   */
   onSuggestedQueryChange?(q: string): Promise<unknown> | void;
 }
 
+/**
+ * Event sent from the registered query suggestions to the parent search box.
+ */
 export type SearchBoxSuggestionsEvent = (
+  /**
+   * The bindings passed from the search box to the suggestions.
+   */
   bindings: SearchBoxSuggestionsBindings
 ) => SearchBoxSuggestions;
 
+const searchBoxElements = ['atomic-search-box', 'atomic-insight-search-box'];
+
+/**
+ * The bindings passed from the search box to the suggestions.
+ */
 export interface SearchBoxSuggestionsBindings extends Bindings {
+  /**
+   * The unique id of the search box.
+   */
   id: string;
+  /**
+   * Whether the search box is standalone.
+   */
   isStandalone: boolean;
+  /**
+   * The SearchBox headless controller.
+   */
   searchBoxController: SearchBox;
+  /**
+   * The amount of queries displayed when the user interacts with the search box.
+   * By default, a mix of query suggestions and recent queries will be shown.
```

Comment:
I would clarify that this is a limit imposed by the `numberOfQueries` prop on the search box, which will limit the number of suggestions displayed but ignore those that don't have a `query` property.

Reply:
What do you mean by "but ignore those that don't have a `query` property?

Reply:
A given suggestions "renderer" renders multiple suggestions, and each can have or not have a `query` property independently. A good example is the recent queries suggestions, which render a bunch of suggestions that each have their own `query` property and also renders a header than doesn't have a `query` property. The header doesn't count towards `numberOfQueries`, but the suggestions below it do.

Reply:
Thanks!